src/libs/actions/Report.js

```diff
@@ -754,16 +759,17 @@ function fetchAllReports(
             // We are waiting 8 seconds since this provides a good time window to allow the UI to finish loading before
             // bogging it down with more requests and operations.
             }, shouldDelayActionsFetch ? 8000 : 0);
-
+        })
+        .finally(() => {
             // Update currentlyViewedReportID to be our first reportID from our report collection if we don't have
             // one already.
             if (!shouldRedirectToReport || lastViewedReportID) {
                 return;
             }
```

Comment:
Maybe consider changing the `shouldRedirectToReport` parameter. As you can see from this check - it will not be respected if `lastViewedRportID` is alredy set. 
Initially I thought that calling `fetchAllReports(shouldRedirectToReport: true)` will redirect to the report no matter what
But now it seems like it only serves to prevent a redirect

Instead of adding flags to the function you can do something like this in the only 2 usages:

##### Report.js
```js
NetworkConnection.onReconnect(() => {
    fetchAllReports();
});
```

##### AuthScreens
```jsx
class AuthScreens extends React.Component {
  componentDidMount() {
    this.initReports();
  }

  initReports() {
    fetchAllReports()
      .then(reportIds => {
         if (!this.props.lastViewedReportID) {
            const firstReportID = _.first(reportIDs);
            updateCurrentlyViewedReportID(firstReportID);
         }
      })
  }
}
```

I guess the concierge creating logic can be extracted as well.

What I'm saying is `fetchAllReports` is not only responsible for fetching reports but at least 2 other things.
If it's intended to always create the Concierge chat when there are no chats it can be renamed to something like `initUserReports` so the purpose is clear. From the current documentation the purpose of the function is just fetching

Even if the function is reused on the 10 places a specific usage like in the AuthScreens example can be extracted instead of flag parameters

Reply:
I've actually found a different way and addressed this